Source/WebCore/dom/ViewTransition.h

```diff
@@ -127,14 +127,27 @@ struct OrderedNamedElementsMap {
         return nullptr;
     }
 
+    void swap(OrderedNamedElementsMap& other)
+    {
+        m_keys.swap(other.m_keys);
+        m_map.swap(other.m_map);
+    }
+
 private:
     ListHashSet<AtomString> m_keys;
     HashMap<AtomString, UniqueRef<CapturedElement>> m_map;
 };
 
+struct ViewTransitionParams {
+    OrderedNamedElementsMap namedElements;
+    FloatSize initialLargeViewportSize;
+    float initialPageZoom;
+};
+
 class ViewTransition : public RefCounted<ViewTransition>, public CanMakeWeakPtr<ViewTransition>, public ActiveDOMObject {
 public:
-    static Ref<ViewTransition> create(Document&, RefPtr<ViewTransitionUpdateCallback>&&);
+    static Ref<ViewTransition> create(Document&, std::optional<RefPtr<ViewTransitionUpdateCallback>>&&);
```

Comment:
For the sake of clarity, maybe separate same-page & outbound. Having  the `std::optional<RefPtr<ViewTransitionUpdateCallback>>&&` type in the constructor doesn't help readability.

```
static Ref<ViewTransition> createSamePage(Document&, RefPtr<ViewTransitionUpdateCallback>&&);

static Ref<ViewTransition> createOutbound(Document&);

static Ref<ViewTransition> createInbound(Document&, std::unique_ptr<ViewTransitionParams>)
```